http/relay/relay.go

```diff
@@ -224,3 +257,47 @@ func isListening(u *url.URL) bool {
 	conn.Close()
 	return true
 }
+
+// cleanupAbandoned consumes the cmdCleanup channel and signals abandoned processes to shut down
+// and release their resources.
+func (p *Proxy) cleanupAbandoned() {
+	for cmd := range p.cmdCleanup {
+		if cmd == nil {
+			continue
+		}
+
+		// Set up a channel to wait for this process to complete processing cleanly
+		waitDone := make(chan bool, 1)
+		go func() {
+			err := cmd.Wait()
+			ps := cmd.ProcessState
+			if e, ok := err.(*exec.ExitError); ok && e != nil {
+				ps = e.ProcessState
+			}
+
+			exitCode := "?"
+			if ps != nil {
+				sys := ps.Sys()
+				if x, ok := sys.(syscall.WaitStatus); ok {
+					exitCode = fmt.Sprintf("%d", x.ExitStatus())
+				}
+			}
+
+			ctx.Infof("proxy (pid=%d) exited with code=%s", cmd.Process.Pid, exitCode)
+			waitDone <- true
+		}()
+
+		// We have deemed this command suitable for cleanup, but we aren't positive the reason
+		// was because of an actual process shutdown.  First try to nicely send a SIGINT.
+		cmd.Process.Signal(os.Interrupt)
+
+		select {
+		case <-waitDone:
+			continue
+		case <-time.After(10 * time.Second):
+			ctx.Warnf("proxy (pid=%d) sending SIGKILL", cmd.Process.Pid)
+			cmd.Process.Kill()
+			<-waitDone
```

Comment:
No matter what, we have to wait for the `Wait` to return, otherwise we don't know if the resources were atually released.